pkg/innerring/processors/netmap/process_epoch.go

```diff
@@ -65,14 +69,88 @@ func (np *Processor) processNewEpoch(ev netmapEvent.NewEpoch) {
 		}
 	}
 
-	np.netmapSnapshot.update(*networkMap, epoch)
+	if np.netmapSnapshot.update(*networkMap, epoch) {
+		l.Debug("updating placements in Container contract...")
+		err = np.updatePlacementInContract(*networkMap, l)
+		if err != nil {
+			l.Error("can't update placements in Container contract", zap.Error(err))
+		} else {
+			l.Debug("updated placements in Container contract")
+		}
+	}
 	np.handleCleanupTick(netmapCleanupTick{epoch: epoch, txHash: ev.TxHash()})
 	np.handleNewAudit(audit.NewAuditStartEvent(epoch))
 	np.handleAuditSettlements(settlement.NewAuditEvent(epoch))
 	np.handleAlphabetSync(governance.NewSyncEvent(ev.TxHash()))
 	np.handleNotaryDeposit(ev)
 }
 
+func (np *Processor) updatePlacementInContract(nm netmap.NetMap, l *zap.Logger) error {
+	// TODO: https://github.com/nspcc-dev/neofs-node/issues/3045
+	cids, err := np.containerWrp.List(nil)
+	if err != nil {
+		return fmt.Errorf("can't get containers list: %w", err)
+	}
+
+	for _, cID := range cids {
+		l := l.With(zap.String("cid", cID.String()))
+		l.Debug("updating container placement in Container contract...")
+
+		cnr, err := np.containerWrp.Get(cID[:])
+		if err != nil {
+			l.Error("can't get container to update its placement in Container contract", zap.Error(err))
+			continue
+		}
+
+		policy := cnr.Value.PlacementPolicy()
+
+		vectors, err := nm.ContainerNodes(policy, cID)
+		if err != nil {
+			l.Error("can't build placement vectors for update in Container contract", zap.Error(err))
+			continue
+		}
+
+		err = np.putPlacementVectors(cID, policy, vectors)
+		if err != nil {
+			l.Error("can't put placement vectors to Container contract", zap.Error(err))
+			continue
+		}
+
+		l.Debug("updated container placement in Container contract")
```

Comment:
at this point, only the request has been sent, are we sure that it will succeed?

Reply:
idk, it is our common approach. i do not remember that we are waiting for TX in any of our handlers

> are we sure that it will succeed?

it is now easy to wait. not sure we need it. it is a potential blocker. alphabet node did everything it can